diffsync/utils.py

```diff
@@ -42,3 +47,44 @@ def __missing__(self, key):
         """When trying to access a nonexistent key, initialize the key value based on the internal factory."""
         self[key] = value = self.factory()
         return value
+
+
+# from: https://stackoverflow.com/questions/72618673/list-directory-tree-structure-in-python-from-a-list-of-path-file
+def _tree(data: dict, prefix: str = ""):
+    """Given a dictionary will yield a visual tree structure.
+
+    A recursive generator, given a dictionary will yield a visual tree structure line by line
+    with each line prefixed by the same characters.
+    """
+    # contents each get pointers that are ├── with a final └── :
+    pointers = [TEE] * (len(data) - 1) + [LAST]
+    for pointer, path in zip(pointers, data):
+        yield prefix + pointer + path
+        if isinstance(data[path], dict):  # extend the prefix and recurse:
+            extension = BRANCH if pointer == TEE else SPACE
+            # i.e. SPACE because LAST, └── , above so no more |
+            yield from _tree(data[path], prefix=prefix + extension)
+
+
+def tree_string(data: dict, root):
+    """String wrapper around `_tree` function to add header and provide tree view of a dictionary."""
+    return "\n".join([root, *_tree(data)])
+
+
+def set_key(data, keys):
+    """Set a nested dictionary key given a list of keys."""
+    current_level = data
+    for key in keys:
+        current_level = current_level.setdefault(key, {})
```

Comment:
Can we get the whole signature type hinted including the return values here? I have opened #175 to make sure we catch this in the CI